corehq/apps/app_manager/static/app_manager/js/modules/report_module.js

```diff
@@ -394,6 +401,29 @@ hqDefine('app_manager/js/modules/report_module', function () {
         for (i = 0; i < currentReports.length; i += 1) {
             self.reports.push(newReport(currentReports[i]));
         }
+
+        // flag instance ids defined outside this module
+        var uuidsByInstanceId = hqImport('hqwebapp/js/initial_page_data').get('uuids_by_instance_id'),
+            instanceIdsInThisModule = _.map(self.reports(), function (r) {return r.instanceId();}),
+            instanceIdsElsewhere = _.filter(_.keys(uuidsByInstanceId), function (instanceId) {
+                return !_.contains(instanceIdsInThisModule, instanceId);
+            });
+
+        var getDuplicatedSlugs = function () {
+            var instanceIdsInThisModule = _.map(self.reports(), function (r) {return r.instanceId();}),
+                duplicatesWithElsewhere = _.filter(instanceIdsInThisModule, function(iid) {
+                    return _.contains(instanceIdsElsewhere, iid);
+                }),
+                duplicatesHere = _.chain(instanceIdsInThisModule)
```

Comment:
This is a little hard to parse...if `instanceIdsInThisModule` and `instanceIdsElsewhere` are both strings of ids, which is what they look like, couldn't you use `_.intersection(instanceIdsInThisModule, instanceIdsElsewhere)` to return the duplicates with other modules and compare `instanceIdsInThisModule.length` to `_.uniq(instanceIdsInThisModule).length` to see if there are duplicates within this module?

Reply:
I wanted to display to the user the list of duplicated IDs, not just whether or not IDs were duplicated, which makes the code a little more complex.  I merged in https://github.com/dimagi/commcare-hq/pull/20169 , so I'm switching validation to that method anyways.